File: src/main/java/net/galaxycore/citybuild/shop/ShopChunkData.java
package net.galaxycore.citybuild.shop;

import lombok.Getter;
import org.bukkit.Chunk;

import java.io.*;
import java.util.ArrayList;
import java.util.List;

@Getter
public class ShopChunkData {
    private final File file;
    private final Chunk chunk;
    private List<Shop> shopsInThisChunk = new ArrayList<>();

    public ShopChunkData(File file, Chunk chunk) {
        this.file = file;
        this.chunk = chunk;
    }

    public void load() {
        try {
            FileInputStream fileIn = new FileInputStream(file);
            ObjectInputStream in = new ObjectInputStream(fileIn);
            //noinspection unchecked
            shopsInThisChunk = (List<Shop>) in.readObject();
            in.close();
            fileIn.close();
        } catch (IOException | ClassNotFoundException i) {
            i.printStackTrace();
        }
    }

    public void save() {
        if(file.exists()) {
            //noinspection ResultOfMethodCallIgnored
            file.delete();
        }
        try {
            //noinspection ResultOfMethodCallIgnored
            file.createNewFile();
            FileOutputStream fileOut = new FileOutputStream(file);
            ObjectOutputStream out = new ObjectOutputStream(fileOut);
            out.writeObject(shopsInThisChunk);
            out.close();
            fileOut.close();
        } catch (IOException i) {
            i.printStackTrace();
        }
    }
}

File: src/main/kotlin/net/galaxycore/citybuild/shop/ShopChunkData.kt
package net.galaxycore.citybuild.shop

import lombok.Getter
import org.bukkit.Chunk
import java.io.*

@Getter
class ShopChunkData(private val file: File, private val chunk: Chunk) {
    var shopsInThisChunk: List<Shop> = ArrayList()
    fun load() {
        try {
            val fileIn = FileInputStream(file)
            val inStream = ObjectInputStream(fileIn)
            @Suppress("UNCHECKED_CAST")
            shopsInThisChunk = inStream.readObject() as List<Shop>
            inStream.close()
            fileIn.close()
        } catch (i: IOException) {
            i.printStackTrace()
        } catch (i: ClassNotFoundException) {
            i.printStackTrace()
        }
    }

    fun save() {
        if (file.exists()) {
            file.delete()
        }
        try {
            file.createNewFile()
            val fileOut = FileOutputStream(file)
            val out = ObjectOutputStream(fileOut)
            out.writeObject(shopsInThisChunk)
            out.close()
            fileOut.close()
        } catch (i: IOException) {
            i.printStackTrace()
        }
    }
}